src/main/java/com/github/distributedtracing/ContinuationLocal.java

```diff
@@ -0,0 +1,170 @@
+package com.github.distributedtracing;
+
+public final class ContinuationLocal<T> {
+    private static InheritableThreadLocal<Object[]> localCtx =
```

Comment:
It does seem InheritableThreadLocal leads to unpredictability. For instance, `ThreadPoolExecutor.execute()` does create threads, which would lead to the threads in the pool each having a "random" value in the `ContinuationLocal`. It would seem better to obviously break so that the user discovers the missing `wrap()`.

There are also issues with object retention, in that the ThreadPoolExecutor, even when properly wrapped, would be retaining references to the `ContinuationLocal` information from "arbitrary" threads, some of which may be long-dead.